system-addon/lib/TopSitesFeed.jsm

```diff
@@ -73,6 +77,10 @@ this.TopSitesFeed = class TopSitesFeed {
           this.refresh(action);
         }
         break;
+      case at.OPEN_NEW_WINDOW:
+      case at.OPEN_PRIVATE_WINDOW:
+        this.openNewWindow(action);
```

Comment:
This seems a little dangerous to rely on the caller to set `isPrivate`. We can just have the `PRIVATE` case `action = Object.assign({}, action, {isPrivate: true})` and add a comment to explicitly fall through.